### PR TITLE
LPS-143347 Must not delete object field of relationship type

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://issues.liferay.com/browse/LPS-143347

This is a follow-up from an observation made by Brian when he reviewed: https://github.com/brianchandotcom/liferay-portal/pull/111592.

The previous work prevents the user to delete an object field of relationship type when using the UI. This work prevents deleting using the Rest API.

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
LGTM :D

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Hey @gabrielwas, could you review my solution for the failing tests?
The solution:
When we delete a relationship, the object field relationship type must be deleted. So, I call the other delete method that doesn't check the type of the field (deletes immediately). The tests are passing with this solution.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java [hunk 1]
```diff
@@ -127,6 +127,11 @@ public ObjectField deleteObjectField(long objectFieldId)
 		ObjectField objectField = objectFieldPersistence.findByPrimaryKey(
 			objectFieldId);
 
+		if (Validator.isNotNull(objectField.getRelationshipType())) {
+			throw new ObjectFieldRelationshipTypeException.
+				MustNotDeleteObjectFieldRelationshipType();
+		}
+
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:134]: @marcelabc this check should be inside the `deleteObjectField(ObjectField objectField)`. That way, we ensure that we don't overpass this validation when someone deletes any field.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 2]
```diff
@@ -166,7 +166,8 @@ public ObjectRelationship deleteObjectRelationship(
 				ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {
 
 			_objectFieldLocalService.deleteObjectField(
-				objectRelationship.getObjectFieldId2());
+				_objectFieldLocalService.getObjectField(
+					objectRelationship.getObjectFieldId2()));
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:170]: @marcelabc that way, we should rethink this solution. I have a suggestion. Let's talk later.

**Comment by marcelabc** [comment]
ci:test:sf

**Pull request closed without merging**